certificates/install_windows.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Arduino SA
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// code inspired by https://github.com/FiloSottile/mkcert licenced under BSD3
+
+package certificates
+
+import (
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+	"unsafe"
+
+	"github.com/arduino/go-paths-helper"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	modcrypt32                           = syscall.NewLazyDLL("crypt32.dll")
+	procCertAddEncodedCertificateToStore = modcrypt32.NewProc("CertAddEncodedCertificateToStore")
+	procCertCloseStore                   = modcrypt32.NewProc("CertCloseStore")
+	procCertOpenSystemStoreW             = modcrypt32.NewProc("CertOpenSystemStoreW")
+)
+
+// // InstallCertificate will install the certificates in the system keychain on windows
+func InstallCertificate(certFile *paths.Path) {
+	// Load cert
+	cert, err := ioutil.ReadFile(certFile.String())
+	if err != nil {
+		log.Errorf("failed to read root certificate: %s", err)
+	}
+	// Decode PEM
+	if certBlock, _ := pem.Decode(cert); certBlock == nil || certBlock.Type != "CERTIFICATE" {
+		log.Error("invalid PEM data: decode pem")
+	} else {
+		cert = certBlock.Bytes
+	}
+	// Open root store
+	store, err := openWindowsRootStore()
+	if err != nil {
+		log.Errorf("cannot open root store %s", err)
```

Comment:
Maybe it would be better to open a dialog with an error, or something like that

Reply:
same for macos

Reply:
Done in [a78e9a5](https://github.com/arduino/arduino-create-agent/pull/778/commits/a78e9a515ed6d8704962700e969f7797dd9546c5)